source/mongodb-handshake/handshake.rst

```diff
@@ -70,13 +70,15 @@ Connection handshake
 --------------------
 
 MongoDB uses the ``hello`` or ``isMaster`` commands for handshakes and topology
-monitoring. ``hello`` is the modern and preferred command. ``isMaster`` is referred
-to as "legacy hello" and is maintained for backwards compatibility with servers
+monitoring. ``hello`` is the modern and preferred command. ``hello`` must
+always be sent using the ``OP_MSG`` protocol. ``isMaster`` is referred to as
+"legacy hello" and is maintained for backwards compatibility with servers
 that do not support the ``hello`` command.
 
 If a `server API version <../versioned-api/versioned-api.rst>`__ is requested,
-drivers MUST use the ``hello`` command for the initial handshake. If server API
-version is not requested, drivers MUST use legacy hello for the initial handshake
+drivers MUST use the ``hello`` command for the initial handshake and use the
+``OP_MSG`` protocol. If server API version is not requested, drivers MUST
+use legacy hello for the initial handshake, use the ``OP_QUERY`` protocol,
```

Comment:
"initial handshake" -> "first message of the handshake". This needs to be really clear. Once the driver knows that it is talking to a 3.6+ server, which it will know from the reply to the first message, it must use OP_MSG for the rest of the handshake.

Reply:
Done.